tests/rptest/tests/write_caching_fi_e2e_test.py

```diff
@@ -0,0 +1,241 @@
+    def test_crash_all_with_consumer_group(self):
+        """
+        Similar to the regular `test_crash_all_or_leader` test but with a
+        consumer group. In this test we will restart the consumer in addition to
+        redpanda nodes and verify that the consumer group is able to recover
+        from the data loss event if it previously committed offsets that are
+        now lost.
+        """
+        consumer = KgoVerifierConsumerGroupConsumer(
+            self.test_context,
+            self.redpanda,
+            self.topic_name,
+            msg_size=512,
+            readers=1,
+            loop=False,
+            continuous=True,
+            tolerate_data_loss=True,
+            group_name="test-group",
+        )
+
+        # Ensure the node is clean before starting the test.
+        # We will skip cleaning in all subsequent restarts to retain all the logs.
+        consumer.clean()
+
+        num_restart_rounds = 2
+        num_msg_per_round = 10000
+
+        total_produced = 0
+        total_lost = 0
+        cumulative_lost_offsets = collections.Counter({})
+        prev_hwm = 0
+        for round_ix in range(0, num_restart_rounds + 1):
+            if round_ix > 0:
+                self._crash_and_restart_cluster()
+                self._wait_for_leader()
+                hwm = next(self.rpk.describe_topic(self.topic)).high_watermark
+                lost = num_msg_per_round - (hwm - prev_hwm)
+                prev_hwm = hwm
+
+                self.logger.info(f"Lost messages: {lost} in round {round_ix}")
+                total_lost += lost
+
+            consumer.start(clean=False)
+
+            # It is important to start the consumer before the producer otherwise
+            # metadata requests might trigger a linearizable barrier and flush to disk
+            # which wouldn't interfere with our data-loss testing.
+            KgoVerifierProducer.oneshot(self.test_context, self.redpanda,
+                                        self.topic_name, self.MSG_SIZE,
+                                        num_msg_per_round // 2)
+            total_produced += num_msg_per_round // 2
+
+            # Wait for the consumer to catch up. This ensures it did join the
+            # group and done with the metadata requests which may trigger a
+            # linearizable barrier and flush to disk.
+            consumer.wait_total_reads(num_msg_per_round // 2,
+                                      timeout_sec=120,
+                                      backoff_sec=3)
+
+            # Produce another round, these will likely get lost.
+            KgoVerifierProducer.oneshot(self.test_context, self.redpanda,
+                                        self.topic_name, self.MSG_SIZE,
+                                        num_msg_per_round // 2)
+            total_produced += num_msg_per_round // 2
+
+            # Wait for the consumer to catch up.
+            consumer.wait_total_reads(num_msg_per_round // 2,
```

Comment:
should this be num_msg_per_round // 2 or num_msg_per_round?

Reply:
Oh. Indeed. Fixed.